pkg/packagekit/updates.jsx

```diff
@@ -1064,38 +1066,110 @@ class OsUpdates extends React.Component {
         this._mounted = false;
     }
 
-    checkNeedsRestart(state) {
+    checkNeedsRestart() {
         this.setState({ checkRestartRunning: true });
-        python.spawn(callTracerScript, null, { err: "message", superuser: "require" })
+        return python.spawn(callTracerScript, undefined, { err: "message", superuser: "require" })
                 .then(output => {
+                    debug("tracer succeeded, output:", output);
                     const restartPackages = JSON.parse(output);
                     // Filter out duplicates
                     restartPackages.reboot = [...new Set(shortenCockpitWsInstance(restartPackages.reboot))];
                     restartPackages.daemons = [...new Set(shortenCockpitWsInstance(restartPackages.daemons))];
                     restartPackages.manual = [...new Set(shortenCockpitWsInstance(restartPackages.manual))];
-                    const nextState = { checkRestartAvailable: true, checkRestartRunning: false, restartPackages };
-                    if (state)
-                        nextState.state = state;
-
-                    this.setState(nextState);
+                    debug("tracer parsed restartPackages:", JSON.stringify(restartPackages));
+                    this.setState({ checkRestartAvailable: true, checkRestartRunning: false, restartPackages });
                 })
                 .catch((exception, data) => {
-                    // common cases: this platform does not have tracer installed
-                    if (!exception.message?.includes("ModuleNotFoundError") &&
+                    // tracer not installed? then fall back to dnf needs-restarting
+                    if (exception.message?.includes("ModuleNotFoundError") ||
                         // or supported (like on Arch)
-                        !exception.message?.includes("UnsupportedDistribution") &&
-                        // or polkit does not allow it
-                        exception.problem !== "access-denied" &&
+                        exception.message?.includes("UnsupportedDistribution")) {
+                        debug('tracer not installed:', JSON.stringify(exception), "trying dnf needs-restarting");
+                        return this.checkDnfNeedsRestarting();
```

Comment:
These 3 added lines are not executed by any test.

Reply:
This is all because `dnf needs-restarting` is only a thing on RHEL 10, not on Fedora 41. So it is covered, just not in /devel..